lib/pbench/server/api/__init__.py

```diff
@@ -113,7 +113,7 @@ def register_endpoints(api, app, config):
         Login, f"{base_uri}/login", resource_class_args=(config, logger, token_auth),
     )
     api.add_resource(
-        ForgotPassword, f"{base_uri}/forgot_password", resource_class_args=(config, logger),
+        ForgotPassword, f"{base_uri}/forgot_password", resource_class_args=(config, logger, token_auth),
```

Comment:
We will actually have two API endpoints, right?  One for requesting a reset-token (e.g., `/forgot_password`) and another for using the token to request a reset (e.g., `/reset_password`), right?  (I.e., this code defines `/forgot_password` while the code at line 74 of `lib/pbench/server/api/resources/users_api.py` is referencing `/reset_password`, but I don't see the definition of it here.)

`/forgot_password` should not require an authorization token; `/reset_password` _should_ require a token, but perhaps it will be passed as a path parameter instead of in the header?

Reply:
Could/should these two endpoints instead be implemented as different methods on the _same_ (or related) endpoint(s)?  (E.g., `GET` for "forgot" (with the email as a query parameter) and `PUT` for reset, or `POST` to `/forgot_password` for requesting the token and `PUT` to `/forgot_password/<token>` for reset?)